src/app/src/components/AnimatedArcsAndMap/useCreateArcPath.tsx

```diff
@@ -0,0 +1,128 @@
+import { useCallback } from 'react';
+import { useMap } from 'react-leaflet';
+import L, { LatLngLiteral, LatLngTuple } from 'leaflet';
+import { useGetSpendingOverTimeQuery } from '../../api';
+import { MonthlySpendingOverTime } from '../../types/api';
+import { PROGRESS_FINAL_STEP } from '../../util';
+import { DC_CENTER, MONTHLY_TIME_DURATION } from '../../constants';
+import { StateFeature } from '../states.geojson';
+
+import polylabel from 'polylabel';
+
+export default function useCreateArcPath(
+    arcPathsReference: React.MutableRefObject<Record<string, L.Curve>>
+) {
+    const { data: spending } = useGetSpendingOverTimeQuery();
+    const map = useMap();
+    !map.getPane('arcPathsPane') && map.createPane('arcPathsPane');
+
+    return useCallback(
+        (event: any) => {
+            const state = event.sourceTarget.feature.properties.STUSPS;
+            if (arcPathsReference.current[state] || !spending) {
+                return;
+            }
+            const polygonCenterTuple: LatLngTuple = findPoleofInaccessibility(
+                event.sourceTarget.feature
+            );
+            const midpoint: L.LatLngTuple =
+                getBezierOffsetLatLng(polygonCenterTuple);
+
+            const { startMonth, playDuration } = getPathAnimationValues({
+                spendingForState: spending.find(
+                    data => data.shape_code === state
+                )!,
+                totalTimeSteps: PROGRESS_FINAL_STEP,
+            });
+
+            arcPathsReference.current[state] = new L.Curve(
+                ['M', DC_CENTER, 'Q', midpoint, polygonCenterTuple],
+                {
+                    color: '#2051FF',
+                    weight: 1,
+                    pane: 'arcPathsPane',
+                    animate: {
+                        duration: playDuration,
+                        iterationStart: startMonth,
+                        iterations: 1,
+                        easing: 'ease-out',
+                    },
+                }
+            );
+        },
+        [arcPathsReference, spending]
```

Comment:
I like how this function has turned out. A lot of complicated things are happening here, but they're separated into functions that have a clear, single purpose. In addition, the variable and function names you chose are meaningful and help the reader determine where to look if they need to dive deeper. Nice work!